Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -1301,6 +1301,30 @@ public class ProgramBuilder {
         perform(EndIf())
     }
 
+    public struct SwitchBuilder {
+        public typealias SwitchCaseGenerator = () -> ()
+        fileprivate var defaultCaseGenerator: SwitchCaseGenerator? = nil
+        fileprivate var caseGenerators: [(value: Variable, fallsthrough: Bool, body: SwitchCaseGenerator)] = []
+
+        public mutating func addDefault(_ generator: @escaping SwitchCaseGenerator) { defaultCaseGenerator = generator }
+        public mutating func add(_ v: Variable, fallsThrough: Bool = false, body: @escaping SwitchCaseGenerator) {
+            caseGenerators.append((v, fallsThrough, body))
+        }
+    }
+
+    public func beginSwitch(on v: Variable, body: (inout SwitchBuilder) -> ()) {
```

Comment:
Nit: maybe just call this `switch` since there is now no longer and `endSwitch` or similar?

Reply:
the compiler complains that `switch` is a keyword and if must use it then I need to wrap it in backticks. Thought it's just cleaner to use `beginSwitch` instead but I'm happy to change it :)

Reply:
Ah, I see, how about `doSwitch`? That'd be consistent with other methods, e.g. [doBreak](https://github.com/googleprojectzero/fuzzilli/blob/cb05c8f8205e9f175956005907e856360e5bcc2e/Sources/Fuzzilli/Core/ProgramBuilder.swift#L1334)